dashboard/app/views/devise/registrations/edit.html.haml

```diff
@@ -6,6 +6,19 @@
 - @page_title = t('activerecord.attributes.user.edit_header')
 - no_email = current_user.no_personal_email? || params[:noEmail] == 'true'
 
+-# We determine if the student is potentially locked by looking at their age
+-# If they are older (or there is no policy for them) they are unlocked
+-# This ignores them being explicitly unlocked by parental permission so we
+-# can show the 'Granted' status of that permission later.
+- potentially_locked = Policies::ChildAccount.underage?(current_user)
+-# The student is in a 'lockout' flow if they are potentially locked out and not unlocked
+- studentInLockoutFlow = potentially_locked && !Policies::ChildAccount::ComplianceState.permission_granted?(current_user)
+-# We need to also account for the case when the US State is not specified
+-# All students are locked out of account settings features until they specify these
+- locked = studentInLockoutFlow || current_user.country_code.nil? || current_user.us_state.nil?
```

Comment:
```suggestion
- locked = studentInLockoutFlow || ((current_user.country_code.nil? || current_user.us_state.nil?) && !isUSA)
```
We only need to ask for the US State if the request is coming from the USA, right?

Reply:
That makes sense.